pyproject.toml

```
@@ -136,18 +136,34 @@ exclude = '''
 profile = 'black'
 skip_gitignore = true
 
-[tool.flake8]
-max-line-length = 99
-doctests = false
-exclude = ["*build/", "docs/"]
-select = "C,E,F,W,B,B950"
-ignore = "N802,N806,W503,E203"
-per-file-ignores = [
-    "*/__init__.py: F401",
-    "docs/conf.py: E265",
-    "/^\\s*\\.\\. _.*?: http/: E501"
+[tool.ruff]
+line-length = 99
+
+[tool.ruff.lint]
+select = [
+  "F",
+  "E",
+  "C",
+  "W",
+  "B",
+]
+ignore = [
+  "N802",
+  "N806",
+  "E203",
```

I think we can live with [E203](https://docs.astral.sh/ruff/rules/whitespace-before-punctuation/) here.
```suggestion
  "E203"
```

Let's break this PR with the other two and see where we are using bad function/variable names.